### DISCUSSION TITLE
Prompt for password only when root access is necessary

### DISCUSSION THREAD

**Discussion by ss141309** [SubconsciousCompute/meta-package-manager#53] [Ideas]
Hi,
At present the `meta-package-manager` prompts for the root password regardless of whether root elevation is required or not. This can lead to inconvenience on the user's end. Therefore I propose that the project should only ask for the password when it absolutely requires it.

Does this idea sound good? If yes then I would appreciate it you suggest how should I start.

Thank you

**Comment by VoltaireNoir** [comment]
Hello @ss141309 

Yes, this is an undesirable behavior that is currently present and needs to be corrected. If you'd like to work on it, you can proceed by escalating privileges only for specific commands such as install,uninstall and update (and any other command that requires it). 

@dilawar there's no other particular reason as to why this behavior exists, correct?

**Reply by dilawar** [reply]
Don't escalate sudo if it is not needed! `

Also note that there are multiple crates `sudo2` and https://crates.io/crates/karen . I'd suggest that you use karen or our own fork (https://github.com/dilawar/sudo2). 

Also we don't handle `http_proxy` and `https_proxy` and `no_proxy` environment variables. Perhaps we should copy these to `sudo` environment as well.

**Reply by ss141309** [reply]
thanks, I will start work on this once my midsems get over (i.e. in about 10 days)
